src/thor/bidirectional_astar.cc

```diff
@@ -324,11 +326,21 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
   float sortcost =
       newcost.cost + astarheuristic_forward_.Get(t2->get_node_ll(meta.edge->endnode()), dist);
 
+  // There is an edge case where we may encounter only_restrictions with edges being
+  // marked as not_thru.  Basically the only way to get in this area is via this edge
+  // based on the restriction, but it is also marked as not_thru. We set the not_thru
+  // to false here only if allow_restricted_thru is true and we are not restricted.
+  // This bool is only set to true on the 2nd pass in route_action. See the gurka
+  // test allow_restricted_thru
+  bool thru = (pred.not_thru_pruning() || !meta.edge->not_thru());
```

Comment:
minor nit: I think a var name of `not_thru_pruning` would be more appropriate since it reflects the `not_thru` pruning state of the current edge label (i.e whether we're culling-out not_thru edges or not)

Reply:
I would call this not_thru_pruning and not confuse the issue with the restriction name.